doc/src/usage.rst

```diff
@@ -72,3 +74,41 @@ To enable locked builds, set the ``locked`` entry in ``pyproject.toml``:
 
     - run ``pdm build --locked``
     - set ``PDM_BUILD_LOCKED`` env var to ``true``
+
+Activate the plugin in build backends
+=====================================
+
+You can even use this plugin without PDM. This is enabled by build backend hooks.
+
+Currently, both `pdm-backend <https://backend.pdm-project.org>` and `hatchling <https://hatch.pypa.io>` are supported.
+
+pdm-backend
+~~~~~~~~~~~
+
+.. code-block::
+    :caption: pyproject.toml
+
+    [project]
+    dynamic = ["optional-dependencies"]
```

Comment:
@frostming it seems marking `optional-dependencies` `dynamic` is in fact not needed and the backend plugin works fine without it. Was this intended?

I also see quite a bit of code regarding optional-dependencies in the backend plugin, so I'm assuming yes

Reply:
It's recommended, and enforced by hatchling.

In PEP 621, a property field is either strictly static, or dynamic which shouldn't be specified at the same time.